mount.go

```diff
@@ -16,5 +16,78 @@
 
 package virtcontainers
 
+import (
+	"fmt"
+	"path/filepath"
+	"syscall"
+)
+
 // These mounts need to be created by the agent within the VM
 var systemMounts = []string{"/proc", "/dev", "/dev/pts", "/dev/shm", "/dev/mqueue", "/sys", "/sys/fs/cgroup"}
+
+func major(dev uint64) int {
+	return int((dev >> 8) & 0xfff)
+
+}
+
+func minor(dev uint64) int {
+	return int((dev & 0xff) | ((dev >> 12) & 0xfff00))
+}
+
+type device struct {
+	major      int
+	minor      int
+	mountPoint string
+}
+
+// getDeviceForPath gets the underlying device containing the file specified by path.
+// The device type constitutes the major-minor number of the device and the dest mountPoint for the device
```

Comment:
Could you give an example of what this function is supposed to do ? Given path `/foo/bar/` what would be the device ?
For example, let's say that `/dev/sda1` is mounted on `/a/b/c/`. If I pass `/a/b/c/file1` to this routine it should return:
```Go
device{
     major: major(/dev/sda3)
     minor: minor(/dev/sda3)
     mountPoint: /a/b/c/
}
```

Correct ?